pkg/open/open_linux.go

```diff
@@ -38,28 +56,114 @@ func browser(url string) error {
 	return nil
 }
 
-// getXWaylandAuthority retrieves the X authority file path from
-// the running XWayland process environment.
-func getXWaylandAuthority() (xAuthorityPath string, err error) {
-	xWaylandProcess, err := platform.GetProcessesByName("Xwayland")
+// getProcessesByName returns a list of processes matching the provided executable path running as the
+// current user.
+func getProcessesByName(exePath string) ([]*gopsutil_process.Process, error) {
+	processes, err := platform.GetProcessesByName(path.Base(exePath))
 	if err != nil {
-		return "", fmt.Errorf("get process by name: %w", err)
+		return nil, err
 	}
-	executablePath, err := xWaylandProcess[0].Exe()
-	if err != nil {
-		return "", fmt.Errorf("get executable path: %w", err)
+
+	var out []*gopsutil_process.Process
+	myUid := int32(os.Getuid())
```

Comment:
You will need to dismiss the lint warning, e.g.:
https://github.com/fleetdm/fleet/blob/142c9e3d5b92ec971f18c5e30395d5ba56ba6a54/cmd/fleet/cron.go#L1442

Am guessing this is a-ok given this is a uid number.

Reply:
Yes, that's correct. UIDs are never negative on any modern POSIX system that I'm aware of.